strawberry/relay/fields.py

```diff
@@ -0,0 +1,553 @@
+import dataclasses
+import inspect
+import sys
+from typing import (  # type: ignore[attr-defined]
+    Any,
+    Awaitable,
+    Callable,
+    Dict,
+    ForwardRef,
+    Iterable,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Set,
+    Type,
+    TypeVar,
+    Union,
+    _eval_type,
+    cast,
+    overload,
+)
+from typing_extensions import Literal, get_args, get_origin
+
+from strawberry.annotation import StrawberryAnnotation
+from strawberry.arguments import StrawberryArgument
+from strawberry.field import _RESOLVER_TYPE, StrawberryField
+from strawberry.lazy_type import LazyType
+from strawberry.permission import BasePermission
+from strawberry.type import StrawberryList, StrawberryOptional, StrawberryType
+from strawberry.types.fields.resolver import StrawberryResolver
+from strawberry.types.info import Info
+from strawberry.types.types import TypeDefinition
+from strawberry.utils.await_maybe import AwaitableOrValue
+from strawberry.utils.cached_property import cached_property
+
+from .types import Connection, GlobalID, Node, NodeType
+
+_T = TypeVar("_T")
+
+
+class RelayField(StrawberryField):
+    """Base relay field, containing utilities for both Node and Connection fields."""
+
+    default_args: Dict[str, StrawberryArgument]
+
+    def __init__(self, *args, **kwargs):
+        default_args = getattr(self.__class__, "default_args", {})
+        if isinstance(default_args, dict):
+            self.default_args = default_args.copy()
+        base_resolver = kwargs.pop("base_resolver", None)
+        super().__init__(*args, **kwargs)
+        if base_resolver:
+            self.__call__(base_resolver)
+
+    @property
+    def arguments(self) -> List[StrawberryArgument]:
+        args = {
+            **self.default_args,
+            **{arg.python_name: arg for arg in super().arguments},
+        }
+        return list(args.values())
+
+    @cached_property
+    def is_basic_field(self):
+        return False
+
+    @cached_property
+    def is_optional(self):
+        return isinstance(self.type, StrawberryOptional)
+
+    @cached_property
+    def is_list(self):
+        type_ = self.type
+        if isinstance(type_, StrawberryOptional):
+            type_ = type_.of_type
+
+        return isinstance(type_, StrawberryList)
+
+    def copy_with(
+        self,
+        type_var_map: Mapping[TypeVar, Union[StrawberryType, type]],
+    ) -> "RelayField":
+        # We should use Self in the return value of copy_with,
+        # but mypy is not accepting it
+        retval = cast(RelayField, super().copy_with(type_var_map))
+        retval.default_args = self.default_args
+        return retval
+
+
+class NodeField(RelayField):
+    """Relay Node field.
+
+    Do not instantiate this directly. Instead, use `@relay.node`
```

Comment:
is there any example of this?

Reply:
You got me! hahaha

This is a very dated comment inherited from my code in strawberry-django-plus. Originally this field had the possibility to receive a resolver, but soon after I did that I noticed that it doesn't make sense to have one, because it is just a matter of creating a `@strawberry.field` with a single argument of `id: GlobalID` in it and removed the functionality.

I'll remove this part of the doc